libraries/ESP8266WiFi/src/WiFiClientSecureBearSSL.cpp

```diff
@@ -362,6 +362,19 @@ int WiFiClientSecureCtx::read(uint8_t *buf, size_t size) {
   return 0; // If we're connected, no error but no read.
 }
 
+// return a pointer to available data buffer (size = availableForPeek())
+// semantic forbids any kind of read() before calling peekConsume()
+const char* WiFiClientSecureCtx::peekBuffer ()
+{
+    return (const char*)_recvapp_buf;
+}
+
+// consume bytes after use (see peekBuffer)
+void WiFiClientSecureCtx::peekConsume (size_t consume)
+{
+    br_ssl_engine_recvapp_ack(_eng, consume);
+}
```

Comment:
Here I think there is aproblem.  When BSSL recvapp_ack is called, the object's recvapp_buf/len is no longer valid  (See line 353 above in the original source).  I believe you need to ack the consumed bytes and then clear the `_recvapp_buf` and `recvapp_len`.